### DISCUSSION TITLE
mutateFormDataBeforeFill for related items

### DISCUSSION THREAD

**Discussion by OvalMedia** [filamentphp/filament#4703] [Help] [answered]
`mutateFormDataBeforeFill` gives us the possibility to modify data before it is sent to the form.
But how can this be triggered for a related item? I have invoices and invoiceitems. The items are injected into the edit form through a repeater.

I tried to create resources for the invoice item and put this into the EditInvoiceItem:

```
    protected function mutateFormDataBeforeFill(array $data): array
    {
        $data['price'] /= 100;
        return $data;
    }
```

But this never gets called. I also tried to create an accessor (https://laravel.com/docs/9.x/eloquent-mutators) for the invoice item model but that also seems to be bypassed.

How can I fix this pls?

**Comment by danharrin** [comment, marked as answer]
A relationship repeater? This has it's own `mutateRelationshipDataBeforeFillUsing(function ($data) {})` method that you can use. Since relationships are loaded separate to the main model data.

**Reply by OvalMedia** [reply]
Thx, exactly what I needed. Could not find this in the docs though.
